### ISSUE TITLE
Package system in Rockstar

### ISSUE THREAD

**Issue by Suloch** [RockstarLang/rockstar#219]
How about adding a package system in rockstar. So we can implement the array feature like one in python. Write program in C for list operations and use rockstar functions to do the operations. For example:

1. First there would be some syntax to import the package   
`We need <package_name> as <custom_identifier>`

2. Create an instance of what we imported with normal assignment operation
`<new_identifier> is <custom_identifier> `
or
`<new_identifier> is <custom_identifier>  taking <initials_parameters>`

3. Now operate on the instance
`<new_identifier> taking <operation_name> and <operand/s>`

For lists it would look something like   
`We need peace as the saviour`
`the winter is the saviour`
`the winter taking "append" and "okay"`

This code initializes the the winter as a list with 'okay'.
All the other operations can be done similarly.

This can be implemented in the interpreter I made.

**Comment by gaborsch** [comment]
I think there are a couple of new concepts here, each of them worth a bit longer discussion.

First, it was the native library concept, that was out of scope, I think as it would require binary compatibility of all interpreters. That's not doable IMHO, the closest possibility is implementing "stdlib functions" in Rockatar.

The `we need` feature soumds nice to me, this is a syntax sugar, but that's OK.
We haven't had function aliasing (`as` part), but that's doable, too.

 Currently we don't have classes and objects in Rockstar, what you wrote for `is` is the instantiation feature, that doesn't make sense until we define how to handle it.

So, suppose that what you wrote is fine, for system functions. But we will need a similar one for user functions, too.

I think that the main issue is the binding of a piece of code (e.g. function in our case) and a locator (filename, URL, etc)  and the organization of user codes. Maybe we need a.location mechanism,.but that's another open question.

Bottom line: I support the packaging system, I support the Stdlibs and the user functions, but we need to think it over carefully.

**Comment by Suloch** [comment]
I was thinking of making an package system for 
rockstar functions and then making a library in c to provide wrappers over c functions to make them usable as rockstar functions. Chirp currently supports user functions and I think packing a set of functions written in rockstar won't be a big deal(though might not efficient).

**Comment by gaborsch** [comment]
Rockstar has different implementations in different programming languages (mostly platforms-agnostic). Adding binaries to it makes no sense, ot at least extreme work for all implementations,

Adding some Rockstar code makes much more sense - though not that effeicient, that's right. But efficiency was not among the goals of Rockstar, so we can live with it, I think.

But we do need the packages and imports.
